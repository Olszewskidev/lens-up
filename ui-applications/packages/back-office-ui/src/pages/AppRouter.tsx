import { createBrowserRouter } from "react-router-dom";
import HomePage from "./Home/HomePage";
import LoginPage from "./Login/LoginPage";
import { AppRoutes } from "../utils/constants"

const AppRouter = createBrowserRouter([
    {
        path: AppRoutes.Default,
        errorElement: <div>Error page</div>,
        children: [
            {
                path: AppRoutes.Default,
                element: <LoginPage />
            },
            {
                path: AppRoutes.Login,
                element: <LoginPage />
            },
            {
                path: AppRoutes.Home,
                element: <HomePage />
            },
        ],
    },
]);

export default AppRouter