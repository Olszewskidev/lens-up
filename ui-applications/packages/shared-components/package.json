{
  "name": "@lens-up/shared-components",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "main": "./src/index.ts",
  "scripts": {
    "build": "tsc && vite build",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview"
  },
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "@typescript-eslint/eslint-plugin": "^7.2.0",
    "@typescript-eslint/parser": "^7.2.0",
    "@vitejs/plugin-react-swc": "^3.5.0",
    "eslint": "^8.57.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.6",
    "typescript": "^5.2.2",
    "vite": "^5.2.0"
  },
  "jest": {
    "moduleNameMapper": {
         "\\.(jpg|ico|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
         "\\.(css|less)$": "<rootDir>/mocks/fileMock.js"
       }
   }
}
