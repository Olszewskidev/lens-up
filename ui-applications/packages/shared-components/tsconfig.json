{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "module": "es2020",
    "target": "es2020",
    "moduleResolution": "bundler"
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules"],
  "references": [{ "path": "./tsconfig.node.json" }],
  "jest": {
    "moduleNameMapper": {
         "\\.(jpg|ico|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
         "\\.(css|less)$": "<rootDir>/mocks/fileMock.js"
       }
   }
}