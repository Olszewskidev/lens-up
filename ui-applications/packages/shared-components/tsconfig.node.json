{
  "compilerOptions": {
    "target": "es2020",
    "composite": true,
    "skipLibCheck": true,
    "module": "es2020",
    "moduleResolution": "bundler",
    "allowSyntheticDefaultImports": true,
    "strict": true
  },
  "include": ["vite.config.ts"]
}
