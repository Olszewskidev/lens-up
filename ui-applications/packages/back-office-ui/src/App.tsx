import { RouterProvider } from 'react-router-dom'
import './App.css'
import AppRouter from './pages/AppRouter'

function App() {
  return (
    <RouterProvider router={AppRouter} />
  )
}

export default App
