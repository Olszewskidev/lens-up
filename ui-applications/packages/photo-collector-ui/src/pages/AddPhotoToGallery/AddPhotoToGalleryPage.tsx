import { ChangeEvent, useState } from "react";
import { useAddPhotoToGalleryMutation } from "../../services/PhotoCollectorApi";
import { AddPhotoForm, AddPhotoFormSkeleton } from "./components"
import { ErrorCardComponent, SuccessAlertComponent } from "@lens-up/shared-components";
import { useParams } from "react-router-dom";

const AddPhotoToGalleryPage = () => {
    const { enterCode } = useParams();
    const [addPhoto, { isError, isSuccess, isUninitialized, isLoading }] = useAddPhotoToGalleryMutation();
    const [photo, setPhoto] = useState<File>();

    const handlePhotoInputChange = (event: ChangeEvent<HTMLInputElement>) => {
        // TODO: Validation
        const fileList = event.target.files
        const isEmpty = !fileList || fileList.length === 0;
        if (isEmpty) {
            return
        }
        const photo = fileList[0];
        setPhoto(photo)
    };

    const handleFormSubmit = () => {
        // TODO: Validation
        if (!photo || !enterCode) {
            return
        }
        const formData = new FormData();
        formData.append('File', photo);
        addPhoto({ enterCode, formData })
    }

    return (
        <>
            {
                isUninitialized && <AddPhotoForm
                    handleFormSubmit={handleFormSubmit}
                    handlePhotoInputChange={handlePhotoInputChange}
                    photo={photo}
                    isLoading={isLoading} />
            }
            {
                isError && <ErrorCardComponent />
            }
            {
                isSuccess && <SuccessAlertComponent />
            }
        </>
    )
}

export default AddPhotoToGalleryPage
