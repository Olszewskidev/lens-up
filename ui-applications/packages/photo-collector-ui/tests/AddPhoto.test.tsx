import { render, waitFor, cleanup, screen, getByText, getByLabelText, fireEvent, getByPlaceholderText } from "@testing-library/react"
import { expect, test, describe, vi, beforeAll, afterAll, afterEach } from 'vitest';
import { HttpResponse, http } from "msw";
import { setupServer } from "msw/node";
import AddPhotoToGalleryPage from "../src/pages/AddPhotoToGallery/AddPhotoToGalleryPage"
import { store } from "../src/app/store";
import { Provider } from "react-redux";
import { BrowserRouter, Navigate, Route, RouterProvider, Routes } from "react-router-dom";
import userEvent from '@testing-library/user-event'
import App from "../src/App";
import AppRouter from "../src/pages/AppRouter";
import { ErrorCardComponent, SuccessCardComponent } from "@lens-up/shared-components";

const photo = new File(["happy"], "happy.png", { type: "image/png" });

let handlers = [
    http.post(`${import.meta.env.VITE_PHOTO_COLLECTOR_SERVICE_URL}/upload-photo/0`, async () => {
        return new Response(null, {
            status: 200,
            headers: {
                Allow: 'GET,HEAD,POST',
            },
        })
    })
]

const successCardRender = render(
    <SuccessCardComponent title="Congratulations!" text="You just joined to the party." />
).asFragment();

/**
* @vitest-environment jsdom
*/
describe("Add photo to gallery page tests", () => {
    const server = setupServer(...handlers);

    beforeAll(() => {
        server.listen()
    })
    afterEach(cleanup);
    afterAll(() => {
        server.close()
    })

    test("Photo form should be uninitialized", async () => {
        const { asFragment } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <AddPhotoToGalleryPage />
                </BrowserRouter>
            </Provider>);

        expect(asFragment()).toMatchSnapshot();
    })

    test("Photo must upload and render success card", async () => {
        URL.createObjectURL = vi.fn();
        console.log(global.window.location.href);
        const { container, baseElement, getByPlaceholderText, getByRole, asFragment } = render(

            <Provider store={store}>
                <BrowserRouter>
                    <Routes>
                        <Route path="/upload-photo/:enterCode" element={<AddPhotoToGalleryPage />} />
                    </Routes>
                    <Navigate to="/upload-photo/0" replace={true} />
                </BrowserRouter>
            </Provider>

        );
        console.log(global.window.location.href);

        const element = container.querySelector("#dropzone-file") as HTMLElement;

        fireEvent.change(element, { target: { files: [photo] } });

        fireEvent.change(screen.getByPlaceholderText("Your name"), { target: { value: "author" } });

        fireEvent.change(screen.getByPlaceholderText("..."), { target: { value: "happy" } });

        fireEvent.submit(screen.getByRole('button'));

        let successRender = asFragment();

        await waitFor(() => {
            successRender = asFragment();
            expect(successRender).toEqual(successCardRender);
        }).then(() => expect(successRender).toMatchSnapshot());
    })
});

const errorCardRender = render(
    <ErrorCardComponent />
).asFragment();

describe("Add photo to gallery page tests unsuccessfully", () => {

    handlers = [
        http.post(`${import.meta.env.VITE_PHOTO_COLLECTOR_SERVICE_URL}/upload-photo/0`, async () => {
            return new Response(null, {
                status: 400,
                headers: {
                    Allow: 'GET,HEAD,POST',
                },
            })
        })
    ]

    const server = setupServer(...handlers);

    beforeAll(() => {
        server.listen()
    })
    afterEach(cleanup);
    afterAll(() => {
        server.close()
    })

    test("Photo must upload and render error card", async () => {
        URL.createObjectURL = vi.fn();
        const { container, baseElement, getByPlaceholderText, getByRole, asFragment } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Routes>
                        <Route path="/upload-photo/:enterCode" element={<AddPhotoToGalleryPage />} />
                    </Routes>
                    <Navigate to="/upload-photo/0" replace={true} />
                </BrowserRouter>
            </Provider>
        );

        const element = container.querySelector("#dropzone-file") as HTMLElement;

        fireEvent.change(element, { target: { files: [photo] } });

        fireEvent.change(screen.getByPlaceholderText("Your name"), { target: { value: "author" } });

        fireEvent.change(screen.getByPlaceholderText("..."), { target: { value: "happy" } });

        fireEvent.submit(screen.getByRole('button'));

        let errorRender = asFragment();

        await waitFor(() => {
            errorRender = asFragment();
            expect(errorRender).toEqual(errorCardRender);
        }).then(() => expect(errorRender).toMatchSnapshot());
    })
});