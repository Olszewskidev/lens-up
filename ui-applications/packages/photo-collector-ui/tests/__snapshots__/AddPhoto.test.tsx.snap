// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add photo to gallery page tests > Photo form should be uninitialized 1`] = `
<DocumentFragment>
  <div
    class="container w-full mx-auto items-center py-32"
  >
    <div
      class="max-w-sm mx-auto overflow-hidden items-center"
    >
      <div>
        <img
          alt="LensUp logo."
          height="147"
          src="/@fs/C:/Users/safronov/source/lens-up/ui-applications/packages/shared-components/src/images/lens-up-logo.png"
          width="379"
        />
        <form
          class="px-7 py-4 bg-black rounded-lg leading-none border-double border-4 border-white shadow-lg max-w-sm mx-auto"
        >
          <div
            class="px-4 py-6"
          >
            <div
              class="mb-5 items-center mx-auto text-center cursor-pointer "
            >
              <label
                class="flex flex-col items-center justify-center w-full"
                for="dropzone-file"
              >
                <div
                  class="flex flex-col items-center justify-center text-white"
                >
                  <svg
                    class="w-8 h-8 mb-4"
                    fill="none"
                    height="24"
                    stroke="#ffffff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="18"
                      rx="2"
                      width="18"
                      x="3"
                      y="3"
                    />
                    <circle
                      cx="8.5"
                      cy="8.5"
                      r="1.5"
                    />
                    <path
                      d="M20.4 14.5L16 10 4 20"
                    />
                  </svg>
                  <p
                    class="mb-2 text-sm"
                  >
                    <span
                      class="font-semibold"
                    >
                      Click to upload your photo
                    </span>
                  </p>
                </div>
                <input
                  class="hidden"
                  id="dropzone-file"
                  type="file"
                />
              </label>
            </div>
            <div
              class="mb-5"
            >
              <div>
                <label
                  class="block mb-2 text-sm font-medium text-gray-50 dark:text-white"
                  for=":r0:"
                >
                  Author:
                </label>
                <input
                  class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  id=":r0:"
                  placeholder="Your name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="mb-5"
            >
              <div>
                <label
                  class="block mb-2 text-sm font-medium text-gray-50 dark:text-white"
                  for=":r1:"
                >
                  Your wishes:
                </label>
                <textarea
                  class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  id=":r1:"
                  placeholder="..."
                  required=""
                  rows="4"
                />
              </div>
            </div>
            <div
              class="flex items-center justify-center"
            >
              <div
                class="text-center relative group"
              >
                <button
                  class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-full text-sm w-full sm:w-auto px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 inline-flex items-center justify-center"
                  type="submit"
                >
                  Upload
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Add photo to gallery page tests > Photo must upload and render success card 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden items-center"
  >
    <div
      class="px-16 py-14"
    >
      <div
        class="flex justify-center"
      >
        <div
          class="rounded-full p-6 bg-green-200"
        >
          <div
            class="flex h-16 w-16 items-center justify-center rounded-full p-4 bg-green-500"
          >
            <svg
              class="h-8 w-8 text-white"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.5 12.75l6 6 9-13.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <h3
        class="my-4 text-center text-3xl font-semibold text-gray-700"
      >
        Congratulations!
      </h3>
      <p
        class="w-[230px] text-center font-normal text-gray-600"
      >
        You just joined to the party.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Add photo to gallery page tests unsuccessfully > Photo must upload and render error card 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow-md overflow-hidden items-center"
  >
    <div
      class="px-16 py-14"
    >
      <div
        class="flex justify-center"
      >
        <div
          class="rounded-full p-6 bg-red-200"
        >
          <div
            class="flex h-16 w-16 items-center justify-center rounded-full p-4 bg-red-500"
          >
            <svg
              class="h-8 w-8 text-white"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.5 12.75l6 6 9-13.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <h3
        class="my-4 text-center text-3xl font-semibold text-gray-700"
      >
        Error!
      </h3>
      <p
        class="w-[230px] text-center font-normal text-gray-600"
      >
        Something went wrong, sorry.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
